File: clients/ios/Lextures/Core/LMS/LMSFeatureModelsArchivedCoursesAdmin.swift
import Foundation

// MARK: - Global archived courses admin (M14.10)

struct ArchivedCourseRow: Decodable, Identifiable, Hashable {
    var id: String
    var courseCode: String
    var title: String
    var archivedAt: String?
    var archivedByUserId: String?
    var archivedByName: String?
    var archivedByEmail: String?
}

struct ArchivedCoursesListResponse: Decodable {
    var courses: [ArchivedCourseRow]
}

File: clients/android/app/src/main/kotlin/com/lextures/android/core/lms/LmsFeatureModelsArchivedCoursesAdmin.kt
package com.lextures.android.core.lms

import kotlinx.serialization.Serializable

@Serializable
data class ArchivedCourseRow(
    val id: String,
    val courseCode: String,
    val title: String,
    val archivedAt: String? = null,
    val archivedByUserId: String? = null,
    val archivedByName: String? = null,
    val archivedByEmail: String? = null,
)

@Serializable
data class ArchivedCoursesListResponse(
    val courses: List<ArchivedCourseRow> = emptyList(),
)